functions/packages.bash

```diff
@@ -284,9 +285,8 @@ mqtt_setup() {
     if ! cond_redirect sed -i -e '/allow_anonymous/d' /etc/mosquitto/mosquitto.conf; then echo "FAILED"; return 1; fi
     if cond_redirect rm -f /etc/mosquitto/passwd; then echo "OK"; else echo "FAILED"; return 1; fi
   fi
-  cond_redirect mkdir -p /var/log/mosquitto
-  cond_redirect chown mosquitto /var/log/mosquitto
   grep -qs "/log.bind" /etc/ztab && if ! zram_dependency install mosquitto; then return 1; fi
+  if ! mkdir -p /opt/zram/log.bind/mosquitto /var/log/mosquitto && chown mosquitto /opt/zram/log.bind/mosquitto /var/log/mosquitto; then echo "FAILED (create zram logdir)"; return 1; fi
```

Comment:
My one concern is that I have Mosquitto installed and running on my personal machine and never had an issue with it not starting, What could be making it not work now? Same for all of the other instances of this as well, why have we never observed this behavior before now, especially because I have personally seen them work.

Reply:
See https://github.com/openhab/openhabian/issues/1538#issuecomment-863225019
Me neither but I and probably you as well we have manually installed mosquitto. Once the logdir on the lower FS is created it'll work from there on. But that doesn't happen in a fresh image based install.

Reply:
My question is why is this a bug that keeps recurring, my only guess is that somehow the overlayfs-tools are not properly handling the directory creation in the log dir, but I'm not sure why. For now it should work as a solution but I am still wanting to find the cause of the problem.